api/v1beta2/clusterclass_types.go

```diff
@@ -98,17 +98,10 @@ type ClusterClassSpec struct {
 	// +kubebuilder:validation:MaxItems=32
 	AvailabilityGates []ClusterAvailabilityGate `json:"availabilityGates,omitempty"`
 
-	// infrastructure is a reference to a provider-specific template that holds
-	// the details for provisioning infrastructure specific cluster
-	// for the underlying provider.
-	// The underlying provider is responsible for the implementation
-	// of the template to an infrastructure cluster.
+	// infrastructure is a reference to a local struct that holds the details
+	// for provisioning the infrastructure cluster for the Cluster.
 	// +optional
-	Infrastructure LocalObjectTemplate `json:"infrastructure,omitempty"`
-
-	// infrastructureNamingStrategy allows changing the naming pattern used when creating the infrastructure object.
-	// +optional
-	InfrastructureNamingStrategy *InfrastructureNamingStrategy `json:"infrastructureNamingStrategy,omitempty"`
+	Infrastructure InfrastructureClass `json:"infrastructure,omitempty"`
```

Comment:
I know this should be a pointer, but I chose to be consistent with the controlPlane field below for now.

Added a sub-task to the v1beta2 umbrella issue to follow-up

Reply:
We need to make sure that we correct this before we ship.

`InfrastructureClass` is a struct, and cannot be `omitempty`'d correctly.

`LocalObjectTemplate` is inlined and has a required field within.

The required field should not accept `""`, which means a structured client marshalling this right now creates an invalid object and cannot treat `Infrastructure` as truly optional

Reply:
I agree. I just don't want to create a different set of problems between this field and the ControlPlane field

Because even if we make the ControlPlane field a pointer it's still not perfect as the required Ref field in there is a pointer

Reply:
Yep, that's fine, is the umbrella issue tracking what's required and what's optional to be completed before the next release?

Reply:
My personal goal would be to fix all these marshalling problems if somehow possible once the KAL linter is available. 

Independent of that. I don't know if this should be a showstopper for the release, given we have the exact same situation for the existing ControPlane field

Reply:
The optionalfields work for KAL I think is pretty close. Might be able to get that here next week, and at least it will highlight where we have marshalling issues. Fixes will still be partially manual/need to be discussed

Reply:
If it helps for the current PR. I think the PR doesn't make it worse as we had the same problem on this field before? (probably? :))